core/vm/runtime/runtime.hpp

```diff
@@ -129,7 +131,13 @@ namespace fc::vm::runtime {
      * Get Message for actor invocation
      * @return message invoking current execution
      */
-    virtual std::shared_ptr<UnsignedMessage> getMessage() = 0;
+    virtual std::reference_wrapper<const UnsignedMessage> getMessage() = 0;
+
+    virtual outcome::result<void> chargeGas(const BigInt &amount) = 0;
+
+    virtual ActorSubstateCID getHead() = 0;
```

Comment:
Please, add description like `Returns Actor State head`. Consider renaming to something more meaningful `getActorStateHead`.